packages/core/useMemoize/index.ts

```diff
@@ -0,0 +1,118 @@
+import { reactive } from 'vue-demi'
+
+type CacheKey = any
+
+/**
+ * Custom memoize cache handler
+ */
+export interface MemoizeCache<Key, Value> {
+  /**
+   * Get value for key
+   */
+  get (key: Key): Value | undefined
+  /**
+   * Set value for key
+   */
+  set (key: Key, value: Value): void
+  /**
+   * Return flag if key exists
+   */
+  has (key: Key): boolean
+  /**
+   * Delete value for key
+   */
+  delete (key: Key): void
+  /**
+   * Clear cache
+   */
+  clear (): void
+}
+
+/**
+ * Memoized function
+ */
+export interface MemoizedFn <Result, Args extends unknown[]> {
+  /**
+   * Get result from cache or call memoized function
+   */
+  (...args: Args): Result
+  /**
+   * Call memoized function and update cache
+   */
+  load (...args: Args): Result
```

Comment:
Do you think we should support async? Or explicit `asyncGet` and `asyncLoad`? I think for many cases, the things we want to cache are time or computation heavy tasks, many of them might be async.

Reply:
It is already possible to pass an async function. In this case the promise gets cached and will be returned no matter if still pending or fulfilled. I did not at the `async` so it can also be used for non async functions.